### PR TITLE
feat: Add eth_getProof RPC

### PR DESCRIPTION
**Pull request by spartucus**
This PR add `eth_getProof` RPC as in https://github.com/appliedzkp/zkevm-circuits/issues/177.

It uses latest `web3` code instead of released crate, we should open an issue to track this after merged.

It also uses ignored unit test case, like other RPC test cases, better fix them after https://github.com/appliedzkp/zkevm-circuits/pull/202 merged.

### CODE REVIEW

File: bus-mapping/Cargo.toml
Comment (CPerezz) [inline bus-mapping/Cargo.toml:17]: The releasing cycle of `rust-web3` is quite fast. Is `Proof` the only thing we need from the new version?

File: bus-mapping/src/rpc.rs
Comment (CPerezz) [inline bus-mapping/src/rpc.rs:246]: Unsure why you don't have the full hash values here. 

How can the assertion suceed if we recieve the full hashes from the RPC call but here we declare them on this way?

**Review by CPerezz — COMMENTED** [review]
Some questions.

But looks promising. Thanks a lot @spartucus :heart:

File: bus-mapping/Cargo.toml
Comment (spartucus) [reply bus-mapping/Cargo.toml:17]: Yes, just need new version.

File: bus-mapping/src/rpc.rs
Comment (spartucus) [reply bus-mapping/src/rpc.rs:246]: It is taken from https://eips.ethereum.org/EIPS/eip-1186, need to fix it with some real data after #202 merged.

File: bus-mapping/Cargo.toml
Comment (ed255) [reply bus-mapping/Cargo.toml:17]: Currently we're using `web3` for the types and `ethers` for the json-rpc; but `ethers` also provides the ethereum types.  In https://github.com/appliedzkp/zkevm-circuits/pull/201 I migrated the types from `web3` to `ethers` to remove the dependency to `web3`.  I see that `ethers` has the Proof type already in a release: https://docs.rs/ethers-core/0.6.1/ethers_core/types/struct.StorageProof.html

So in case #201 is merged first, I think it would be nice if you can update this PR to use the `StorageProof` type from `ethers` :)

File: bus-mapping/src/rpc.rs
Comment (ed255) [reply bus-mapping/src/rpc.rs:246]: I'd prefer to have a valid test vector before merging this PR.  I understand that maybe it's better to wait for #202 so that we can have a reproducible test :)

File: bus-mapping/Cargo.toml
Comment (CPerezz) [reply bus-mapping/Cargo.toml:17]: Ohh I completely forgot. Sorry @ed255  my bad. 

Ye, in that case I completely agree with Edu's point here. Let's wait till #201 gets merged and then use the new types! :)

**Comment by CPerezz** [comment]
@spartucus #201 has been merged. So you can go for it. 

 I recommend to first rebase so that no conflicts arise. :)

**Comment by ed255** [comment]
> @spartucus #201 has been merged. So you can go for it.
> 
> I recommend to first rebase so that no conflicts arise. :)

+1

File: bus-mapping/Cargo.toml
Comment (spartucus) [reply bus-mapping/Cargo.toml:17]: sounds good!:)

File: bus-mapping/src/rpc.rs
Comment (spartucus) [reply bus-mapping/src/rpc.rs:246]: Sure thing:)

**Review by CPerezz — APPROVED** [review]
LGTM

Wait for a review from @ed255 and once approved, we can merge I think! :)

**Review by ed255 — APPROVED** [review]

**Comment by ed255** [comment]
LGTM! Thanks @spartucus :)
I will merge this PR once all the tests pass

**Pull request merged**
